app/js/RequestLogger.js

```diff
@@ -1,32 +1,83 @@
 const logger = require('logger-sharelatex')
+const metrics = require('metrics-sharelatex')
 
 module.exports = {
-  logRequest,
-  logError
+  errorHandler,
+  middleware
 }
 
-function logRequest(req, res) {
-  // response has already been sent, but we log what happened here
-  logger.log(
-    {
-      info: res.logInfo,
-      url: req.originalUrl,
-      params: req.params
-    },
-    res.logMsg || 'HTTP request'
-  )
+function errorHandler(err, req, res, next) {
+  req._logInfo.set('error', err)
+  res
+    .send(err.message)
+    .status(500)
+    .end()
 }
 
-function logError(err, req, res, next) {
-  logger.err(
-    {
-      err,
-      info: res.logInfo,
-      url: req.originalUrl,
-      params: req.params,
-      msg: res.logMsg
-    },
-    err.message
-  )
-  next(err) // use the standard error handler to send the response
+function middleware(req, res, next) {
+  const startTime = new Date()
+
+  // methods to allow the setting of additional information to be logged for the request
+  req._logInfo = {}
+  req._logMessage = 'http request'
+  req.addLogField = function(field, value) {
+    req._logInfo[field] = value
+  }
+  req.addLogFields = function(fields) {
+    Object.assign(req._logInfo, fields)
+  }
+  req.setLogMessage = function(message) {
+    req._logMessage = message
+  }
```

Comment:
A little code organization suggestion:

`_logInfo`, `_logMessage`, `addLogFields` and `setLogMessage` implement together the logging feature. How about bringing them together in an object?
```
class RequestLogger {
  constructor() {
    this._logInfo = {}
    this._logMessage = 'http request'\
  }
  addFields(...) { ... }
  setMessage(..) { ... }
  emitLog(...) {
    logger.log(this._logMessage, ...)
  }
}

function middleware(req, res, next) {
  ...
  req.logger = new RequestLogger()
  ...
  res.end = function() {
    ...
    req.logger.emitLog()
    ...
  }
}
```